src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1877,7 +1912,12 @@ def _load_checkpoint_to_model(self):
                 )
 
         # UPDATE TRAINING PARAMS IF THEY EXIST & WE ARE NOT LOADING AN EXTERNAL MODEL's WEIGHTS
-        self.best_metric = self.checkpoint["acc"] if "acc" in self.checkpoint.keys() else -1
+        _best_ckpt_metrics = get_param(self.checkpoint, "_best_ckpt_metrics")
+        if self.load_checkpoint or resume_path:
```

Comment:
Are `load_checkpoint` or `resume_path` synonims? Why are we using the same flow for both these cases? IMO - when resuming we load everything and resuming experiment (in same folder).
When `load_checkpoint` - we simply load weights to a model, but starting from scratch and not looking at the old weights. Am I making a right assumption here?

Reply:
The `if self.load_checkpoint or resume_path:` can indeed be simplified to `if self.load_checkpoint`.
Loading weights only is not handled since long time in the Trainer itself, but through models.get(), which is used by Trainer and one can pass certain weights through checkpoint_params. 
Long story short- the load_checkpoint attribute is used to load weights for resuming training.